src/serviceWorker.ts

```diff
@@ -0,0 +1,127 @@
+import firebase from 'firebase'
+
+export const service_worker = ()=>{
+    if ("serviceWorker" in navigator) {
+        navigator.serviceWorker.register("./sw.js").then(() => {
+          //Service Worker Registration Succesful
+          console.log('[Service Worker] Registration : Successful');
+
+            navigator.geolocation.getCurrentPosition(showPosition=>{
+              const {latitude , longitude} = showPosition.coords 
+              // console.log(latitude , longitude)
+              fetch(`https://api.bigdatacloud.net/data/reverse-geocode-client?latitude=${latitude}&longitude=${longitude}&localityLanguage=en`)
+              .then(res => res.json() )
+              .then(data =>{
+                const loc = data.localityInfo.administrative.slice(-1)[0].name.split(' ')[0]
+                localStorage.setItem("userLocation" , JSON.stringify(loc))
+              })
+              .catch(err => console.log("[Location] : Permission not granted"))
+            });
+        
+        })
+        .catch(err=>{
+          console.log('[Service Worker] Registration : Failed');
+        })
+      }
+      if ("Notification" in window) {
+ 
+          Notification.requestPermission((result) => {
+            if (result === "granted") 
+              {
+        
+                ConfigurePushSubscription()
+              }
+          });
+
+      }
+
+}
+
+export default service_worker
+
+
+
+
+
+////////////Default noti
+
+const confirmationNotification = () => {
+  if (!("serviceWorker" in navigator))
+  return;
+navigator.serviceWorker.ready.then((activeSW) => {
+const NotificationOptions = {
+body: "We will notify you with the new posts in your city !!",
+};
+activeSW.showNotification(
+"Thanks for enabling notifications!",
+NotificationOptions
+);
+});
+}
+
+
+
+////////Push Configuration
+
+const ConfigurePushSubscription = () => {
```

Comment:
By convention, function name should start with smaller case and be in camelcase only.
Classes names are uppercase